server.js

```diff
@@ -1,35 +1,128 @@
 // import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
 // import topMusicData from './data/top-music.json'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/shows"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+
+  const Show = mongoose.model('Show', {
+    showID:{
+      type: Number,
+    },
+    title: {
+      type: String,
+    },
+    directors: {
+      type: String,
+    },
+    cast: {
+      type: String,
+    },
+    country: {
+      type: String,
+    },
+    date_added: {
+      type: String,
+    },
+    release_year: {
+      type: Number,
+    },
+    rating: {
+      type: String,
+    },
+    duration: {
+      type: String,
+    },
+    listed_in: {
+      type: String,
+    },
+    description: {
+      type: String,
+    },
+    type: {
+      type: String,
+    },
+  })
+
+  if (process.env.RESET_DATABASE) {
+    console.log('Resetting database!')
+    
+    const seedDatabase = async () => {
+    
+      await Show.deleteMany()
+
+      await netflixData.forEach((show) => new Show(show).save())
+    
+    }
+  seedDatabase()
+  }
+
+
+const port = process.env.PORT || 5000
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
 // Start defining your routes here
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('My endpoints= /shows & /shows?query=movie or /shows?query=Tv Show & /shows/:title & /shows/year/:release_year & /shows/country/:country' )
+})
+
+app.get('/shows', async (req, res) => {
+  const { query } = req.query
+   if (query) {
+    //localhost:5000/shows?query=Showtype 
+    const queryRegex = new RegExp(query, 'i')
+    const shows = await Show.find({ type: queryRegex })
+    res.json(shows)
+   } else {
+    //localhost:5000/shows
+    const shows = await Show.find();
+    res.json(shows)
+   }
+ })
+
+app.get('/shows/:title', async (req, res) => {
+  const { title } = req.params
+  const show = await Show.findOne({ title: title })
+  if (show) {
+    res.json(show)
+  } else {
+    res.status(404).json({ error: `Could not find show with title=${title}` })
+  }
+})
+
+app.get('/shows/year/:release_year', async (req, res) => {
+  const { release_year } = req.params
+  const year = await Show.find({ release_year: release_year })
+  if (year) {
+    res.json(year)
+  } else {
+    res.status(404).json({ error: 'Not found' })
+  }
+})
+
+app.get('/shows/country/:country', async (req, res) => {
+  const { country } = req.params
+  const whatCountry = await Show.find({ country: country })
+    res.json(whatCountry)
```

Comment:
Endpoints should be named after the data they are returning, and not the way they are filtering the data. To make it more RESTful you could include these two as queries in you `/shows` endpoint:
`/shows?year=2010&country=sweden`